app/main/helpers/frameworks.py

```diff
@@ -5,6 +7,26 @@
 from werkzeug.datastructures import ImmutableOrderedMultiDict
 
 
+def has_registered_interest_in_framework(client, framework_slug):
+    audits = client.find_audit_events(
+        audit_type=AuditTypes.register_framework_interest.value,
+        object_type='suppliers',
+        object_id=current_user.supplier_id)
+    for audit in audits['auditEvents']:
+        if audit['data'].get('frameworkSlug') == framework_slug:
+            return True
+    return False
+
+
+def register_interest_in_framework(client, framework_slug):
+    client.create_audit_event(
```

Comment:
It gets handled by the try catch over here https://github.com/alphagov/digitalmarketplace-supplier-frontend/pull/170/files#diff-2830492bae5ad218e386be64bc065150R25